classes/class-admin.php

```diff
@@ -728,19 +759,165 @@ public function wp_ajax_reset() {
 	private function erase_stream_records() {
 		global $wpdb;
 
-		$where = '';
+		// If this is a multisite and it's not networked activated,
+		// only delete the entries from the blog which made the request.
+		if ( self::WP_STREAM_MULTI_NOT_NETWORK === $this->plugin->get_site_type() ) {
 
-		if ( is_multisite() && ! $this->plugin->is_network_activated() ) {
-			$where .= $wpdb->prepare( ' AND `blog_id` = %d', get_current_blog_id() );
+			// First check the log size.
+			$stream_log_size = self::get_blog_record_table_size();
+
+			// If this is a large log and we need to delete only the entries
+			// pertaining to an individual site, we will need to do those in batches.
+			if ( $this->plugin->is_large_records_table( (int) $stream_log_size ) ) {
```

Comment:
@tharsheblows The `get_blog_record_table_size` method always returns `int`, there's no need for a cast here:

```suggestion
			if ( $this->plugin->is_large_records_table( $stream_log_size ) ) {
```